File: codility/src/main/java/codility/com/lessons/l04countingelements/MaxCounters.java
package codility.com.lessons.l04countingelements;

/**
 * 100%
 * @author 08115341622
 */
public class MaxCounters {
	public int[] solution(int N, int[] A) {
		int resp[] = new int[N];
		int max = 0;
		int somar = 0;
		int reiniciou = -1;
		for (int i = 0; i < A.length; i++) {
			int x = A[i];
			if (x == N + 1) {
				reiniciou = i;
				somar += max;
				max = 0;
			} else {
				if (reiniciou != -1) {
					resp = new int[N];
					reiniciou = -1;
				}
				resp[x - 1]++;
				if (max < resp[x - 1]) {
					max = resp[x - 1];
				}
			}
		}
		if (reiniciou != -1) {
			resp = new int[N];
			reiniciou = -1;
		}
		for (int i = 0; i < N; i++) {
			resp[i] += somar;
		}
		return resp;
	}
}

File: codility_new/max_counters.py
# https://app.codility.com/programmers/lessons/4-counting_elements/max_counters/

def solution(N, A):
    max_c = N + 1
    count_values = {}
    max_so_far = 0
    default_value = 0
    for command in A:
        if command == max_c:
            count_values.clear()
            default_value = max_so_far
        else:
            count_values[command] = count_values.get(command, default_value) + 1
            max_so_far = max(max_so_far, count_values[command])

    return [count_values.get(i+1, default_value) for i in range(N)]
